torch/nn/functional.py

```diff
@@ -744,14 +744,17 @@ def triplet_margin_loss(anchor, positive, negative, margin=1.0, p=2, eps=1e-6, s
     return loss
 
 
-def normalize(input, p, dim, eps=1e-12):
-    r"""Performs l_p normalization of inputs over specified dimension.
+def normalize(input, p=2, dim=-1, eps=1e-12):
+    r"""Performs :math:`L_p` normalization of inputs over specified dimension.
 
     Does:
 
     .. math::
         v = \frac{v}{\max(\lVert v \rVert_p, \epsilon)}
 
-    for each subtensor v over dimension dim of input.
+    for each subtensor v over dimension dim of input. Each subtensor is flattened into a vector,
+    i.e. :math:`\lVert v \rVert_p` is not a matrix norm.
+
+    With default arguments normalizes over the last dimension with Euclidean norm.
```

Comment:
Wouldn't it make more sense to default to normalization over the 1st (feature) dim? Also, can you add the docstrings for each argument?